consensus/polybft/bridge_event_manager.go

```diff
@@ -0,0 +1,569 @@
+	if err != nil {
+		return fmt.Errorf("failed to generate hash for BridgeBatch. Error: %w", err)
+	}
+
+	hashBytes := hash.Bytes()
+
+	signature, err := b.config.key.SignWithDomain(hashBytes, signer.DomainStateReceiver)
+	if err != nil {
+		return fmt.Errorf("failed to sign batch message. Error: %w", err)
+	}
+
+	sig := &MessageSignature{
+		From:      b.config.key.String(),
+		Signature: signature,
+	}
+
+	sourceChainID := pendingBridgeBatch.BridgeMessageBatch.SourceChainID.Uint64()
+
+	if _, err = b.state.BridgeMessageStore.insertMessageVote(
+		epoch,
+		hashBytes,
+		sig,
+		dbTx,
+		sourceChainID); err != nil {
+		return fmt.Errorf(
+			"failed to insert signature for message batch to the state. Error: %w",
+			err,
+		)
+	}
+
+	// gossip message
+	b.multicast(&TransportMessage{
+		Hash:          hashBytes,
+		Signature:     signature,
+		From:          b.config.key.String(),
+		EpochNumber:   epoch,
+		SourceChainID: sourceChainID,
+	})
+
+	numberOfMessages := len(pendingBridgeBatch.BridgeMessageBatch.Messages)
+	if numberOfMessages > 0 {
+		b.logger.Debug(
+			"[buildBridgeBatch] build batch",
+			"from", pendingBridgeBatch.BridgeMessageBatch.Messages[0].ID.Uint64(),
+			"to", pendingBridgeBatch.BridgeMessageBatch.Messages[numberOfMessages-1].ID.Uint64(),
+		)
+	}
+
+	b.lock.Lock()
+	defer b.lock.Unlock()
+
+	b.pendingBridgeBatches = append(b.pendingBridgeBatches, pendingBridgeBatch)
+
+	return nil
+}
+
+// multicast publishes given message to the rest of the network
+func (b *bridgeEventManager) multicast(msg interface{}) {
+	data, err := json.Marshal(msg)
+	if err != nil {
+		b.logger.Warn("failed to marshal bridge message", "err", err)
+
+		return
+	}
+
+	err = b.config.topic.Publish(&polybftProto.TransportMessage{Data: data})
+	if err != nil {
+		b.logger.Warn("failed to gossip bridge message", "err", err)
+	}
+}
+
+// EventSubscriber implementation
+
+// GetLogFilters returns a map of log filters for getting desired events,
+// where the key is the address of contract that emits desired events,
+// and the value is a slice of signatures of events we want to get.
+// This function is the implementation of EventSubscriber interface
+func (b *bridgeEventManager) GetLogFilters() map[types.Address][]types.Hash {
+	var stateSyncResultEvent contractsapi.StateSyncResultEvent
+
+	return map[types.Address][]types.Hash{
+		contracts.StateReceiverContract: {types.Hash(stateSyncResultEvent.Sig())},
+	}
+}
+
+// ProcessLog is the implementation of EventSubscriber interface,
+// used to handle a log defined in GetLogFilters, provided by event provider
+func (b *bridgeEventManager) ProcessLog(header *types.Header, log *ethgo.Log, dbTx *bolt.Tx) error {
+	var bridgeMessageResultEvent *contractsapi.BridgeMessageResultEvent
+
+	doesMatch, err := bridgeMessageResultEvent.ParseLog(log)
+	if err != nil {
+		return err
+	}
+
+	if !doesMatch {
+		return nil
+	}
+
+	return b.state.BridgeMessageStore.removeBridgeEventsAndProofs(bridgeMessageResultEvent)
```

Comment:
Before doing this, check if the `SourceChainID` of the event matches the `sourceChainID` from the struct.